dipy/core/tests/test_gradients.py

```diff
@@ -27,6 +30,79 @@ def test_unique_bvals_deprecated():
                       np.array([0, 800, 1400, 1401, 1405]))
 
 
+def test_read_bvals_bvecs(tmp_path: pathlib.Path):
+    # NOTE: This data is hard-coded to this particular dataset
+    _, fbvals, fbvecs = get_fnames('small_64D')
+    expected_bvals_shape = (65,)
+    expected_bvec_shape = (65, 3)
+    # Case 1: bval + bvec present
+    case_1_bval, case_1_bvec = read_bvals_bvecs(fbvals=fbvals, fbvecs=fbvecs)
+    assert case_1_bval.shape == expected_bvals_shape
+    assert case_1_bvec.shape == expected_bvec_shape
+    # Case 2: bval only
+    case_2_bval, case_2_bvec = read_bvals_bvecs(fbvals=fbvals, fbvecs=None)
+    assert case_2_bval.shape == expected_bvals_shape
+    assert case_2_bvec is None
+    # Case 3: bvec only
+    case_3_bval, case_3_bvec = read_bvals_bvecs(fbvals=None, fbvecs=fbvecs)
+    assert case_3_bval is None
+    assert case_3_bvec.shape == expected_bvec_shape
+    # Case 4: read seralized numpy array
+    case4_bval_numpy_filename = tmp_path / "bval.npy"
+    case4_bvec_numpy_filename = tmp_path / "bvec.npy"
+    np.save(file=case4_bval_numpy_filename, arr=case_1_bval)
+    np.save(file=case4_bvec_numpy_filename, arr=case_1_bvec)
+    case_4_bval, case_4_bvec = read_bvals_bvecs(
+        fbvals=str(case4_bval_numpy_filename),
+        fbvecs=str(case4_bvec_numpy_filename)
+    )
+    assert case_4_bval.shape == expected_bvals_shape
+    assert case_4_bvec.shape == expected_bvec_shape
+    np.testing.assert_allclose(case_4_bval, case_1_bval)
+    np.testing.assert_allclose(case_4_bvec, case_1_bvec)
+
+    # TODO: Expand this test, not sure what would trigger this
```

Comment:
I was trying to write a test for this section -- https://github.com/dipy/dipy/pull/2589/files#diff-6e52b8b6894f91b9bc345dba41c51a89ee4bdd2f60d24b77cab81a1fce42fcefR77 -- but I couldn't figure out an easy way to trigger it. Probably because I suck at numpy..... Open to suggestions

Reply:
I did not look at the details, so sorry if I am wrong.

It seems you are doing the same as [here](https://github.com/dipy/dipy/blob/master/dipy/io/tests/test_io_gradients.py#L128).

Reply:
Yeah I just removed all my changes to this file as they are duplicated